Utils/LeftAlignedCollectionViewFlowLayout.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  LeftAlignedCollectionViewFlowLayout.swift
+//  Movies
+//
+//  Created by Gabriel Rodrigues Minucci on 29/06/2022.
+//
+
+import UIKit
+
+class LeftAlignedCollectionViewLayout: UICollectionViewFlowLayout {
+    override func layoutAttributesForElements(in rect: CGRect) -> [UICollectionViewLayoutAttributes]? {
+        let attributes = super.layoutAttributesForElements(in: rect)
+
+        var leftMargin = sectionInset.left
+        var maxY: CGFloat = -1.0
+        attributes?.forEach { layoutAttribute in
+            if layoutAttribute.frame.origin.y >= maxY {
+                leftMargin = sectionInset.left
+            }
+
+            layoutAttribute.frame.origin.x = leftMargin
+
+            leftMargin += layoutAttribute.frame.width + minimumInteritemSpacing
+            maxY = max(layoutAttribute.frame.maxY , maxY)
+        }
+
+        return attributes
+    }
+}
```

Comment:
I'm afraid, you've lost me here:

maxY = max(layoutAttribute.frame.maxY , maxY)

 I could use an explanation too🙏

Reply:
I've added comments on this class to explain what is happening there. Feel free to ask more about it in our next session if you still have questions.
In case the question was just about the `max` method, it's a math method that returns the max value between parameters, for example:
```swift
let a = max(3, 4) // a is 4
let a = min(3, 4) // a is 3
```

Reply:
I just didn't get the last line. Now I do! Amazing, love it!